src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -10797,29 +10811,43 @@ def _rolling_agg(
             )
 
         # Throw NotImplementedError if any parameter is unsupported
-        check_is_rolling_window_supported_by_snowflake(rolling_kwargs)
+        if window_func == "rolling":
+            check_is_rolling_window_supported_by_snowflake(window_kwargs)
+        else:
+            check_is_expanding_window_supported_by_snowflake(window_kwargs)
+
         frame = query_compiler._modin_frame.ensure_row_position_column()
         row_position_quoted_identifier = frame.row_position_snowflake_quoted_identifier
         if center:
             # -(window // 2) is equivalent to window // 2 PRECEDING
             rows_between_start = -(window // 2)  # type: ignore
             rows_between_end = (window - 1) // 2  # type: ignore
         else:
-            # 1 - window is equivalent to window - 1 PRECEDING
-            rows_between_start = 1 - window  # type: ignore
+            if window_func == "rolling":
```

Comment:
instead of using string, can you simply define and enum to make all usage defnition centralized

Reply:
Yep done.